### ISSUE TITLE
Get original string

### ISSUE THREAD

**Issue by beltik** [RedMadRobot/input-mask-ios#21]
How to get original string?
For example, i have mask {(123)}-[000] and text field text is "(123)456, i want to get what exactly user enter, is that case - 456.

**Comment by taflanidi** [comment]
Hey @beltik 
Thanks for your question. 

`MaskedTextFieldDelegate` has a listener with a callback method `textField(_:didFillMandatoryCharacters:didExtractValue:)`, where the «extracted value» is a variable you seek.

**Comment by beltik** [comment]
Thanks.
I also want to feature to add any character or digit or symbol to user input.
Currently for do that, i drag library to my object and modified.
Reason is, i  needed symbols like ".%@?" etc.

**Comment by taflanidi** [comment]
@beltik,

I'm not sure I've understood you right, but there's a [feature request](https://github.com/RedMadRobot/input-mask-android/issues/28) about special characters, and the work is in progress. 

P.s. Also, according to your mask format, for `{(123)}-[000]` you will receive a resulting value like `(123)000`, with `(123)` prefix included.

In order to omit it, you should incorporate a format like `(123)-[000]`.

**Comment by beltik** [comment]
@taflanidi 

If i have mask like [-] or [_] i can only enter letters or numbers, but what about symbols like dot? Currently i can't type dot and other symbols in field.

To add that ability i did drag library from pods to project and modify it. There is a method that returns only specific character set depend on enum case, i simply comment it and return YES (because it suite to my needs). But better solution is to add something like [<key>] that allow to edit all symbols.

**Comment by beltik** [comment]
Let me try to explain again. I need ability to add symbols like "," or ".".
For do that, i copied library to my project, and in class ValueState  in func     func accepts(character char: Character) -> Bool {
I replaced code with following:
  case .AlphaNumeric:
//                return CharacterSet.alphanumerics.isMember(character: char)
                return true

But, of course i would preffer adding additional case, like case .everyCharacter.
Hope i make myself clear.

**Comment by taflanidi** [comment]
Implemented in `3.3.0`.
